src/include/gc/gc_manager.h

```diff
@@ -19,6 +19,7 @@
 #include "common/item_pointer.h"
 #include "common/logger.h"
 #include "common/macros.h"
+#include "concurrency/transaction_context.h"
```

Comment:
You're only using a TransactionContext pointer here. You shouldn't need to do this include here. Remove this, fast-forward the declaration of TransactionContext, and put this in the .cpp file.